File: tensor.m
function tensor
    scores = [
        95 83 76
        99 89 82
    ]
    old = scores;
    scores(1, 2) = 85;
    old
    scores
    % mean(scores, 1)
    % mean(scores, 2)
    % flat = scores(:)
    % ndims(scores)
    % ndims(flat)
    % scores(1, 2, 1, 1, 1, 1, 1)
end

File: tensor.py
def main():
    from numpy import array
    from numpy.linalg import matrix_rank

    scores = array([
        [95, 83, 76],
        [99, 89, 82.0],
    ])
    # print(scores.mean(axis=0).ndim)
    # print(scores.mean(axis=1, keepdims=True).ndim)
    # print(scores[0])
    # print(scores[:, ::2])
    # flat = scores.ravel()
    # flat[3] = 97
    # print(flat)
    # print(scores)

    scores[1] = 0.5 * scores[0]
    print(scores)
    print(matrix_rank(scores))


if __name__ == "__main__":
    main()
